cogs/modmail.py

```diff
@@ -0,0 +1,394 @@
+
+        if thread is None:
+            return 
+
+        linked_message_id = None
+            
+        async for msg in ctx.channel.history():
+            if message_id is None and msg.embeds:
+                em = msg.embeds[0]
+                if 'Moderator' not in str(em.footer.text):
+                    continue
+                linked_message_id = int(re.findall(r'\d+', em.author.url)[0])
+                break 
+            elif message_id and msg.id == message_id:
+                url = msg.embeds[0].author.url 
+                linked_message_id = int(re.findall(r'\d+', url)[0])
+                break
+        
+        if not linked_message_id:
+            raise commands.UserInputError
+
+        await thread.edit_message(linked_message_id, new_message)
+        await ctx.message.add_reaction('✅')
+
+    @commands.command()
+    @trigger_typing
+    @commands.has_permissions(manage_channels=True)
+    async def contact(self, ctx, *, user: discord.Member):
+        '''Create a thread with a specified member.'''
+
+        exists = await self.bot.threads.find(recipient=user)
+        if exists:
+            return await ctx.send('Thread already exists.')
+        else:
+            thread = await self.bot.threads.create(user, creator=ctx.author)
+        
+        em = discord.Embed(
+            title='Created thread',
+            description=f'Thread started in {thread.channel.mention} for {user.mention}',
+            color=discord.Color.green()
+            )
+
+        await ctx.send(embed=em)
+    
+    @commands.command()
+    @trigger_typing
+    @commands.has_permissions(manage_channels=True)
+    async def blocked(self, ctx):
+        '''Returns a list of blocked users'''
+        em = discord.Embed(title='Blocked Users', color=discord.Color.green())
+        em.description = ''
+
+        users = []
+        not_reachable = []
+
+        for id in self.bot.blocked_users:
+            user = self.bot.get_user(id)
+            if user:
+                users.append(user)
+            else:
+                not_reachable.append(id)
+        
+        em.description = 'Here is a list of blocked users.'
+        
+        if users:
+            em.add_field(name='Currently Known', value=' '.join(u.mention for u in users))
+        if not_reachable:
+            em.add_field(name='Unknown', value='\n'.join(f'`{i}`' for i in not_reachable), inline=False)
+        
+        if not users and not not_reachable:
+            em.description = 'Currently there are no blocked users'
+
+        await ctx.send(embed=em)
+        
+    @commands.command()
+    @trigger_typing
+    @commands.has_permissions(manage_channels=True)
+    async def block(self, ctx, id=None):
+        '''Block a user from using modmail.'''
+
+        if id is None:
+            thread = await self.bot.threads.find(channel=ctx.channel)
+            if thread:
+                id = thread.recipient.id
```

Comment:
idk i think this is used in other places in code? maybe get a func? 

```suggestion
id = await self.get_current_thread(default=id)
```